### PR TITLE
cmd: refuse --userns if a mapping is specified

### PR DESCRIPTION
**Pull request by giuseppe**
if an explicit mapping is specified, do not accept `--userns` since it
overriden to "private".

Closes: https://github.com/containers/podman/issues/15233

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

#### Does this PR introduce a user-facing change?

```release-note
--userns is no longer accepted when an explicit mapping is specified
```

### CODE REVIEW

File: test/e2e/run_userns_test.go [hunk 1]
```diff
@@ -307,6 +307,25 @@ var _ = Describe("Podman UserNS support", func() {
 		}
 
 	})
+
+	It("podman --userns= conflicts with ui[dg]map and sub[ug]idname", func() {
+		session := podmanTest.Podman([]string{"run", "--userns=private", "--uidmap=0:1:500", "alpine", "true"})
+		session.WaitWithDefaultTimeout()
+		Expect(session).Should(Exit(125))
```
Comment (vrothberg) [inline test/e2e/run_userns_test.go:314]: Could we also check for the error message?

**Review by vrothberg — COMMENTED** [review]
Other than the mini nit: LGTM
@containers/podman-maintainers PTAL

**Review by flouthoc — APPROVED** [review]
LGTM

File: test/e2e/run_userns_test.go (hunk 1 above)
Comment (giuseppe) [reply test/e2e/run_userns_test.go:314]: sure, added now

**Pull request merged**